Index: src/main/java/com/ipa/openapi_inzent/controller/MydataController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/ipa/openapi_inzent/controller/MydataController.java b/src/main/java/com/ipa/openapi_inzent/controller/MydataController.java
--- a/src/main/java/com/ipa/openapi_inzent/controller/MydataController.java	
+++ b/src/main/java/com/ipa/openapi_inzent/controller/MydataController.java	(date 1686829042643)
@@ -435,7 +435,7 @@
     }
     @GetMapping("/provider/customerList")
     @ResponseBody
-    public JsonObject customerList(String customerNum) {
+    public JsonObject customerList(String customerNum, @RequestParam("pageNo") int pageNo) {
         JsonObject object = new JsonObject();
         List<MdProviderDTO> list = mydataService.mdProviderCustomerList(customerNum);
         System.out.println("customerNum = " + customerNum);
